managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java

```diff
@@ -390,4 +376,88 @@ public static void createChecksums() {
       }
     }
   }
+
+  public static X509Certificate getX509CertificateCertObject(String certContent) {
+    try {
+      InputStream in = null;
+      byte[] certEntryBytes = certContent.getBytes();
+      in = new ByteArrayInputStream(certEntryBytes);
+      CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
+      return (X509Certificate) certFactory.generateCertificate(in);
+    } catch (CertificateException e) {
+      LOG.error(e.getMessage());
+      throw new RuntimeException("Unable to get cert Object");
+    }
+  }
+
+  public static PrivateKey getPrivateKey(String keyContent) {
+    try (PemReader pemReader = new PemReader(new StringReader(new String(keyContent)))) {
+      PemObject pemObject = pemReader.readPemObject();
+      byte[] bytes = pemObject.getContent();
+      PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(bytes);
+      KeyFactory kf = KeyFactory.getInstance("RSA");
+      return kf.generatePrivate(spec);
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      throw new RuntimeException("Unable to get Private Key");
+    }
+  }
+
+  public static void writeKeyFileContentToKeyPath(PrivateKey keyContent, String keyPath) {
+    File keyFile = new File(keyPath);
+    try (JcaPEMWriter keyWriter = new JcaPEMWriter(new FileWriter(keyFile))) {
+      keyWriter.writeObject(keyContent);
```

Comment:
Make sure to flush the writer. Same for the Cert.

Reply:
Done. Thanks!